glean-core/ffi/src/lib.rs

```diff
@@ -324,4 +326,67 @@ pub extern "C" fn glean_is_first_run() -> u8 {
     with_glean_value(|glean| glean.is_first_run())
 }
 
+/// A FFI-compatible representation for the PingUploadTask
+#[repr(u8)]
+pub enum FfiPingUploadTask {
+    Wait,
+    Upload {
+        uuid: *const c_char,
```

Comment:
Should all of these be `FfiStr`?

Reply:
Honestly, I didn't because I started having multiple problems with lifetimes when I tried. I didn't go deep enough to understand if I *should* use it here or not. Because I have to return the `FfiPingUploadTask`, the only way I found to do that with the `FfiStr` was to add a 'static lifetime to it, which the documentation strictly says never to do.

The other functions that return strings, also return them as `*mut c_char`, so I decided to also do that for this one. 

I will try and understand if I should or not use `FfiStr` here and try to come back with an answer, but if you or @badboy already know the answer that would be cool too :)

Reply:
`FfiStr` is used to accept strings (and then convert them and help with lifetimes).
On the way out of the FFI we need to use `*const c_char` (actually maybe `*mut c_char`?).

Reply:
Oh and one thing I realize: we need to ensure the platform side frees these strings again (by calling `glean_str_free` or maybe even better by passing back the `FfiPingUploadTask` somehow, so the Rust side can reconstruct the `CString` and free them.

Reply:
> (actually maybe *mut c_char?)

I noticed that is what the other functions return, why is that? FWIW `*const c_char` seems to be working for now.

Reply:
Because it is a mutable pointer after all, as we're giving up ownership (effectively).